internal/pkg/otel/configtranslate/otelconfig.go

```diff
@@ -0,0 +1,366 @@
+	exportersConfig := map[string]any{}
+	exporterType, err := getExporterTypeForComponent(comp)
+	if err != nil {
+		return nil, err
+	}
+	for _, unit := range comp.Units {
+		if unit.Type == client.UnitTypeOutput {
+			unitExportersConfig, expErr := unitToExporterConfig(unit, exporterType)
+			if expErr != nil {
+				return nil, expErr
+			}
+			for k, v := range unitExportersConfig {
+				exportersConfig[k] = v
+			}
+		}
+	}
+	return exportersConfig, nil
+}
+
+// getBeatNameForComponent returns the beat binary name that would be used to run this component.
+func getBeatNameForComponent(comp *component.Component) string {
+	// TODO: Add this information directly to the spec?
+	if comp.InputSpec == nil || comp.InputSpec.BinaryName != "agentbeat" {
+		return ""
+	}
+	return comp.InputSpec.Spec.Command.Args[0]
+}
+
+// getSignalForComponent returns the otel signal for the given component. Currently, this is always logs, even for
+// metricbeat.
+func getSignalForComponent(comp *component.Component) (pipeline.Signal, error) {
+	beatName := getBeatNameForComponent(comp)
+	switch beatName {
+	case "filebeat", "metricbeat":
+		return pipeline.SignalLogs, nil
+	default:
+		return pipeline.Signal{}, fmt.Errorf("input type not supported by Otel: %s", comp.InputType)
+	}
+}
+
+// getReceiverTypeForComponent returns the receiver type for the given component.
+func getReceiverTypeForComponent(comp *component.Component) (otelcomponent.Type, error) {
+	beatName := getBeatNameForComponent(comp)
+	switch beatName {
+	case "filebeat":
+		return otelcomponent.MustNewType(fbreceiver.Name), nil
+	case "metricbeat":
+		return otelcomponent.MustNewType(mbreceiver.Name), nil
+	default:
+		return otelcomponent.Type{}, fmt.Errorf("input type not supported by Otel: %s", comp.InputType)
+	}
+}
+
+// getExporterTypeForComponent returns the exporter type for the given component.
+func getExporterTypeForComponent(comp *component.Component) (otelcomponent.Type, error) {
+	switch comp.OutputType {
+	case "elasticsearch":
+		return otelcomponent.MustNewType("elasticsearch"), nil
+	default:
+		return otelcomponent.Type{}, fmt.Errorf("output type not supported by Otel: %s", comp.OutputType)
+	}
+}
+
+// unitToExporterConfig translates a component.Unit to an otel exporter configuration.
+func unitToExporterConfig(unit component.Unit, exporterType otelcomponent.Type) (map[string]any, error) {
+	if unit.Type == client.UnitTypeInput {
+		return nil, fmt.Errorf("unit type is an input, expected output: %v", unit)
+	}
+	configTranslationFunc, ok := configTranslationFuncForExporter[exporterType]
+	if !ok {
+		return nil, fmt.Errorf("no config translation function for exporter type: %s", exporterType)
+	}
+	// we'd like to use the same exporter for all outputs with the same name, so we parse out the name for the unit id
+	// these will be deduplicated by the configuration merging process at the end
+	outputName := strings.Split(unit.ID, "-")[1]
```

Comment:
Is this the only way to get the output name? What happens if the input already contains a '-' ?

Reply:
The output name is not saved directly anywhere in the component model, so parsing it out of ids is the only way right now. I'd like to explicitly save it in the component model in a follow-up, to avoid any ambiguities.
But yeah, as both the input type and the output name themselves can contain dashes, the only safe way to do this is to explicitly trim the (known) type from the ID.